tests/Browser/Pages/Backend/Categories/AdminListCategoriesTest.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Tests\Browser\Pages\Backend\Categories;
+
+use App\Model\Book;
+use App\Model\Category;
+use App\Model\User;
+use App\Model\Donator;
+use Tests\DuskTestCase;
+use Laravel\Dusk\Browser;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Support\Facades\DB;
+use Faker\Factory as Faker;
+
+class AdminListCategoriesTest extends DuskTestCase
+{
+    use DatabaseMigrations;
+
+    /**
+     * make a User with role 1.
+     *
+     * @return void
+     */
+    public function makeUser(){
+        factory(User::class)->create([
+            'role' => 1
+        ]);
+    }
+
+
+    /**
+     * A Dusk test route to page list categories.
+     *
+     * @return void
+     */
+    public function testRouteShowListCategories()
+    {
+        $this->makeUser();
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs(User::find(1))
+                    ->visit('/admin')
+                    ->clickLink('CATEGORIES')
+                    ->assertPathIs('/admin/categories')
+                    ->assertSee('List Categories');
+        });
+    }
+
+    /**
+     * Test layout of List Categories.
+     *
+     * @return void
+     */
+    public function testLayoutListCategories()
+    {
+        $this->makeUser();
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs(User::find(1))
+                ->visit('/admin/categories')
+                ->assertSee('List Categories')
+                ->assertSeeLink('Admin')
+                ->assertSee('ID')
+                ->assertSee('Name')
+                ->assertSee('Number of Books');
+        });
+    }
+
+    /**
+     * Create virtual database
+     *
+     * @return void
+     */
+    public function makeDataOfListCategories($rows)
+    {
+        $faker = Faker::create();
+
+        factory(Category::class, $rows)->create();
+        $categoryIds = DB::table('categories')->pluck('id')->toArray();
+
+        $donator = factory(Donator::class, 1)->create();
+
+        for ($i = 0; $i <= $rows; $i++)
+        {
+            factory(Book::class,1)->create([
+                'category_id' => $faker->randomElement($categoryIds),
+                'donator_id' => 1,
```

Comment:
ở đậy em tạo $donator thì. em phải 'donator_id' => $donatorIds->id

Reply:
vì chỉ tạo 1 record nên mặc định id of donator là 1